src/common/reducers/entities/index.js

```diff
@@ -0,0 +1,27 @@
+import merge from 'lodash/merge';
+
+import * as RepoActionTypes from '../../actions/repository';
+
+import repository from './repository';
+
+
+// TODO snaps
+
+export function entities(state = { snaps: {}, repos: {} }, action) {
+  if (action.payload && action.payload.entities) {
+    return merge({}, state, action.payload.entities);
+  }
+
+  // only modify repos if action is one of REPO_ types
+  if (action.payload && action.payload.id && RepoActionTypes[action.type]) {
+    return {
+      ...state,
+      repos: {
```

Comment:
I'm not sure if it will be good to have one reducer (one 'slice' of state) to contain all different entities' types from the app.

One point against it is likely weird logic to detect which part of `entities` should be updated by an action (like we do here by checking for `RepoActionTypes`).

Other one is that any component that needs access to any data (so most of them) would need a connect binding to whole `entities`, even if it's interested only in small part of it (for example repos or snaps only).

But let's not worry about it too much now and see how it goes when we add snaps to the equation.